doc/connectivity-semantics-and-api.md

```diff
@@ -28,7 +28,10 @@ through TLS handshake (or equivalent) and all subsequent attempt to communicate
 have succeeded (or are pending without any known failure ).
 
 TRANSIENT_FAILURE: There has been some transient failure (such as a TCP 3-way
-handshake timing out or a socket error). Channels in this state will eventually
+handshake timing out or a socket error). New RPCs will fail immediately with
+status UNAVAILABLE, but implementations may support options like
```

Comment:
Add a similar sentence to explain what happens to new RPCs when channel is in CONNECTING state.